gcc/rust/expand/rust-macro-invoc-lexer.cc

```diff
@@ -49,6 +49,22 @@ MacroInvocLexer::split_current_token (TokenId new_left, TokenId new_right)
 		       std::unique_ptr<AST::Token> (new AST::Token (l_tok)));
 }
 
+void
+MacroInvocLexer::split_current_token (std::vector<TokenPtr> new_tokens)
+{
+  rust_assert (new_tokens.size () > 0);
+
+  auto current_pos = token_stream.begin () + offs;
+
+  token_stream.erase (current_pos);
+
+  for (size_t i = 1; i < new_tokens.size (); i++)
+    {
+      token_stream.insert (current_pos + i, std::unique_ptr<AST::Token> (
```

Comment:
this is gonna start inserting at `current_pos + 1`, but we erase the one at `current_pos`. is that okay and what we want? I can never remember how insertion works in C++